File: Readio/src/main/java/ninegle/Readio/book/domain/Preference.java
package ninegle.Readio.book.domain;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import ninegle.Readio.user.domain.User;

/**
 * Readio - Preferences
 * create date:    25. 5. 13.
 * last update:    25. 5. 13.
 * author:  gigol
 * purpose: 관심목록 엔티티
 */
@Getter
@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)

public class Preference {
	@Override
	public String toString() {
		return "Preference{" +
			"id=" + id +
			", user=" + user +
			", book=" + book +
			'}';
	}

	@Builder
	public Preference(User user, Book book) {
		this.user = user;
		this.book = book;
	}

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;

	@ManyToOne
	@JoinColumn(name = "user_id", nullable = false)
	private User user;

	@ManyToOne
	@JoinColumn(name = "book_id", nullable = false)
	private Book book;



}

File: Readio/src/main/java/ninegle/Readio/book/domain/Preference.kt
package ninegle.Readio.book.domain

import jakarta.persistence.*
import lombok.AccessLevel
import lombok.Builder
import lombok.Getter
import lombok.NoArgsConstructor
import ninegle.Readio.user.domain.User

@Entity
class Preference (
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private val id: Long? = null,

    @field:JoinColumn(
        name = "user_id",
        nullable = false
    )
    @field:ManyToOne
    var user: User,
    @field:JoinColumn(
        name = "book_id",
        nullable = false
    )
    @field:ManyToOne
    var book: Book
) {
    override fun toString(): String {
        return "Preference{" +
                "id=" + id +
                ", user=" + user +
                ", book=" + book +
                '}'
    }


}
